base/strings/search.jl

```diff
@@ -2,23 +2,24 @@
 
 function findnext(pred::EqualTo{Char}, s::String, i::Integer)
     if i < 1 || i > sizeof(s)
-        i == sizeof(s) + 1 && return 0
+        i == sizeof(s) + 1 && return nothing
         throw(BoundsError(s, i))
     end
     @inbounds isvalid(s, i) || string_index_err(s, i)
     c = pred.x
-    c ≤ '\x7f' && return _search(s, c % UInt8, i)
+    c ≤ '\x7f' && return nothing_sentinel(_search(s, c % UInt8, i))
```

Comment:
Is it really useful to change `_search` too, given that it's internal?

Reply:
Doesn't that one always return a range? `_search` caused me more headaches than any other here, so I could be misunderstanding.

Reply:
Ah, right. Though my first remark about changing `_search` still holds I think.

Reply:
Yeah, but I don't feel like doing it :smile:. In truth this is what I tried first, spent half a day just trying to get Julia to build, and then punted and started from scratch. It's possible I was stymied by something very trivial, but at this stage even half a day is not worth it for an internal interface that can be changed later.

I think the problem is that the range-returning version calls the scalar-returning version and I don't know this part of Julia well enough to untangle it without investing more time than I can afford right now.

Reply:
I know that feeling too, that code is complex.

Let's leave it that way, but please move `nothing_sentinel` to this file since it isn't used elsewhere.